Fixes bug 1281540 - Added a new datetime picker for Super Search and Signature report.

adngdb opened the pull request:
cc @lonnen 

Here's what it looks like: 

![](http://adrian.gaudebert.fr/downloads/mozilla/datepicker-signature-report.png)
![](http://adrian.gaudebert.fr/downloads/mozilla/datepicker-super-search.png)

lonnen commented:
@adngdb -- that looks prettier. How does it handle a smaller window? I don't think we need to be fully responsive, but it's common to split a screen in half with browser + terminal or similar. The current layout looks pretty good until you add rules, and then they wrap with no space.

<img width="884" alt="screenshot 2016-09-21 17 50 10" src="https://cloud.githubusercontent.com/assets/21467/18733919/05f39b16-8024-11e6-8f23-278414d7562c.png">

lonnen commented:
Also, moment.js is used all over the webapp and this is a major version bump. The reviewer should check to make sure this will not break other functionality. It may impact [Bug 1304108](https://bugzilla.mozilla.org/show_bug.cgi?id=1304108)

adngdb commented:
@lonnen It looks okay on a smaller screen: 

![](http://adrian.gaudebert.fr/downloads/mozilla/datepicker-super-search-small-screen.png)
![](http://adrian.gaudebert.fr/downloads/mozilla/datepicker-signature-report-small-screen.png)

The dynamic rules are not styled using flex, and I'd love to revisit their style but I thought it was slightly out of scope for this PR.

peterbe commented on webapp-django/crashstats/crashstats/templatetags/jinja_helpers.py:
```diff
@@ -108,6 +108,37 @@ def time_tag(dt, format='%a, %b %d %H:%M %Z', future=False):
+    """
+    return jinja2.Markup('''
+        <span
+            class="datetime-picker {0}"
```
It's a nit but I would use named parameters. E.g
```
"""Foo {name} and {bar} and {name} again""".format(name='name', bar='bar')
```

peterbe commented on webapp-django/crashstats/signature/jinja2/signature/signature_report.html:
```diff
@@ -26,12 +26,35 @@
+            </fieldset>
+
+            <p class="date-shortcuts">
+                Shortcuts, last
```
I would say "Shortcuts: " here instead of the comma.

peterbe commented on webapp-django/crashstats/supersearch/jinja2/supersearch/search.html:
```diff
@@ -25,49 +25,64 @@
+                <a href="#" data-range="1M">1 month</a> -
+                <a href="#" data-range="3M">3 months</a> -
+                <a href="#" data-range="6M">6 months</a>
+            </p>
```
Since this snippet is exactly the same as in `signature_report.html` you should move this into a file and do something like `{% include "crashstats/date-shortcuts.html" %}`

peterbe commented on webapp-django/crashstats/supersearch/static/supersearch/js/socorro/date_filters.js:
```diff
@@ -0,0 +1,93 @@
+        };
+        filters.to.config.onChange = function (dateObj) {
+            filters.from.set('maxDate', dateObj);
+        };
```
Nice!

adngdb commented:
@peterbe comments addressed!

peterbe commented:
When I load the page, the "7 days" shortcut does not have `class="selected"`. 
When I click one of the shortcuts it does set `class="selected"` but there's no CSS that does anything with this.

adngdb commented:
@peterbe I've addressed the missing bits, and while I was at it, I improved the search form UX (it now become visible only when it is fully loaded) and I integrated the loader work of #3484 here. 

@m8ttyB This might change the selector you want to use to get the loader, because I have moved it out of the ``#advanced-search`` element into the ``#search-form`` element. Basically, it is higher up in the DOM tree. This makes the whole form UX much better, and also means that the loader will disappear and be replaced by the entire form only when all of it is ready to be used.

m8ttyB commented:
Adrian,

Thank you for heads up, and yes, I'm excited for this change set. It will
likely make the tests more reliable. I'll update the selectors and include
it in the e2e-tests pr.

~ Matt

On Mon, Sep 26, 2016 at 7:25 PM, Adrian Gaudebert <notifications@github.com>
wrote:

> @peterbe <https://github.com/peterbe> I've addressed the missing bits,
> and while I was at it, I improved the search form UX (it now become visible
> only when it is fully loaded) and I integrated the loader work of #3484
> <https://github.com/mozilla/socorro/pull/3484> here.
>
> @m8ttyB <https://github.com/m8ttyB> This might change the selector you
> want to use to get the loader, because I have moved it out of the
> #advanced-search element into the #search-form element. Basically, it is
> higher up in the DOM tree. This makes the whole form UX much better, and
> also means that the loader will disappear and be replaced by the entire
> form only when all of it is ready to be used.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/socorro/pull/3478#issuecomment-249742789>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAD22WQ4ROU-lPP1W5trgU9ychpMwJ60ks5quHCDgaJpZM4KDZ5X>
> .
>

peterbe commented:
I first clicked on "3 days" then changed my dates manually and it still indicates that it's "3 days"
<img width="993" alt="screen shot 2016-09-29 at 3 01 35 pm" src="https://cloud.githubusercontent.com/assets/26739/18968101/b1531842-8655-11e6-99c2-b597d252bd37.png">

adngdb commented:
@peterbe Good catch! I fixed that in the latest commit.
